### PR TITLE
adding reddit search extension

### PR DESCRIPTION
**Pull request by ewlcheng**
## Description

Searches and displays results from reddit.

You can...

* Do simple quick searches on reddit
* View your search results in different sort orders
* Search for subreddits
* Favorite subreddits for quick access
* Search within subreddits

## Screencast

Example

## Checklist

- [✔️] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [✔️] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [✔️] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/reddit-search/src/Home.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        <>
+          <List.Section title="More ways to search">
+            <List.Item
+              key="searchOnRedditForSubreddits"
+              icon={Icon.MagnifyingGlass}
+              title="Search subreddits..."
+              actions={
+                <ActionPanel>
+                  <Action.Push
+                    title="Search subreddits"
```
Comment (pernielsentikaer) [inline extensions/reddit-search/src/Home.tsx:79]:
```suggestion
                    title="Search Subreddits"
```

File: extensions/reddit-search/src/PostList.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            </ActionPanel>
+          }
+        />
+        <List.Item
+          key="sortByLatest"
+          icon={Icon.MagnifyingGlass}
+          title="Sort by latest"
+          actions={
+            <ActionPanel>
+              <Action title="Sort by latest" onAction={() => doSearch(redditSort.latest)} />
```
Comment (pernielsentikaer) [inline extensions/reddit-search/src/PostList.tsx:86]:
```suggestion
              <Action title="Sort by Latest" onAction={() => doSearch(redditSort.latest)} />
```

File: extensions/reddit-search/src/PostList.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            </ActionPanel>
+          }
+        />
+        <List.Item
+          key="sortByTop"
+          icon={Icon.MagnifyingGlass}
+          title="Sort by top"
+          actions={
+            <ActionPanel>
+              <Action title="Sort by top" onAction={() => doSearch(redditSort.top)} />
```
Comment (pernielsentikaer) [inline extensions/reddit-search/src/PostList.tsx:76]:
```suggestion
              <Action title="Sort by Top" onAction={() => doSearch(redditSort.top)} />
```

File: extensions/reddit-search/src/PostList.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            </ActionPanel>
+          }
+        />
+        <List.Item
+          key="sortByComments"
+          icon={Icon.MagnifyingGlass}
+          title="Sort by comments"
+          actions={
+            <ActionPanel>
+              <Action title="Sort by comments" onAction={() => doSearch(redditSort.comments)} />
```
Comment (pernielsentikaer) [inline extensions/reddit-search/src/PostList.tsx:96]:
```suggestion
              <Action title="Sort by Comments" onAction={() => doSearch(redditSort.comments)} />
```

File: extensions/reddit-search/src/SubredditList.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            actions={
+              <ActionPanel>
+                <Action.Push
+                  title={`Search in r/${x.subredditName}`}
+                  target={<FilterBySubredditPostList subredditName={x.subredditName} subreddit={x.subreddit} />}
+                />
+                <Action.OpenInBrowser url={x.url} icon={Icon.Globe} />
+                {!x.isFavorite && (
+                  <Action
+                    title="Favorite"
```
Comment (pernielsentikaer) [inline extensions/reddit-search/src/SubredditList.tsx:79]:
```suggestion
                    title="Add to Favorites"
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Wow @ewlcheng 🔥 

What an extension, it's massive! 

Some feedback and suggestions for you:

- Is there a way to show 20 items when entering a favorite Subreddit?

- Is there a way to indicate what `sort` that's active?

HENCZrdufP982bmi

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/reddit-search/src/Home.tsx (hunk 1 above)
Comment (ewlcheng) [reply extensions/reddit-search/src/Home.tsx:32]: Done. 👍

File: extensions/reddit-search/src/PostList.tsx (hunk 2 above)
Comment (ewlcheng) [reply extensions/reddit-search/src/PostList.tsx]: Done. 👍

File: extensions/reddit-search/src/PostList.tsx (hunk 3 above)
Comment (ewlcheng) [reply extensions/reddit-search/src/PostList.tsx]: Done. 👍

File: extensions/reddit-search/src/PostList.tsx (hunk 4 above)
Comment (ewlcheng) [reply extensions/reddit-search/src/PostList.tsx]: Done. 👍

File: extensions/reddit-search/src/SubredditList.tsx (hunk 5 above)
Comment (ewlcheng) [reply extensions/reddit-search/src/SubredditList.tsx]: Done. 👍

**Review by pernielsentikaer — COMMENTED** [review]
Hi @ewlcheng 👋 

You're on fire 🔥 

Do you think it's possible to show a list of items when entering a Subreddit? I'd like to see a list like when I use my phone and entering a Subreddit?

**Comment by ewlcheng** [comment]
> Hi @ewlcheng 👋
> 
> You're on fire 🔥
> 
> Do you think it's possible to show a list of items when entering a Subreddit? I'd like to see a list like when I use my phone and entering a Subreddit?

That's a great idea, I like it! Implemented.

File: extensions/reddit-search/src/Home.tsx (hunk 1 above)
Comment (pernielsentikaer) [reply extensions/reddit-search/src/Home.tsx:32]: I think you forgot this one 😄

File: extensions/reddit-search/src/SubredditList.tsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+              <ActionPanel>
+                <Action.Push
+                  title={`Search in r/${x.subredditName}`}
+                  target={<FilterBySubredditPostList subredditName={x.subredditName} subreddit={x.subreddit} />}
+                />
+                <Action.OpenInBrowser url={x.url} icon={Icon.Globe} />
+                {!x.isFavorite && (
+                  <Action
+                    title="Add to Favorites"
+                    icon={Icon.Star}
```
Comment (pernielsentikaer) [inline extensions/reddit-search/src/SubredditList.tsx:100]:
```suggestion
                    icon={Icon.Star}
                    shortcut={{ modifiers: ["cmd"], key: "f" }}
```

File: extensions/reddit-search/src/SubredditList.tsx [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+                        ...results.slice(0, index),
+                        { ...results[index], isFavorite: !results[index].isFavorite },
+                        ...results.slice(index + 1),
+                      ]);
+                    }}
+                  />
+                )}
+                {x.isFavorite && (
+                  <Action
+                    title="Remove from Favorites"
```
Comment (pernielsentikaer) [inline extensions/reddit-search/src/SubredditList.tsx:114]:
```suggestion
                    title="Remove From Favorites"
                    shortcut={{ modifiers: ["cmd"], key: "f" }}
```
Then it's a toggle function 🔥

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Wow, @ewlcheng! 🔥 

You keep surprising! I have some more feedback for you to address, we're so close!

- It seems like there is [Empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states) when entering a `Favorite subreddit` – can you look into this?
- Seems like you had an idea about using a preference value to let the user select the amount of results but didn't finish – is it forgotten or a leftover?
- When entering a favorite subreddit, then it does not make sense to show the subreddit name, does it?

p9YORduWtwbUqc4t

Thanks for creating a very awesome plugin, I'm already using it all the time 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by ewlcheng** [comment]
Good catches @pernielsentikaer. Let's see if I squashed all the bugs this time. :)

**Review by pernielsentikaer — APPROVED** [review]
Awesome @ewlcheng 🔥 

I think we're ready to proceed, let's get it merged, so everyone can use this extension! I'm going to use it a lot myself 👏 

Thanks for your contribution 💯

**Comment by sxn** [comment]
🎉 nice work

**Pull request merged**